lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +213,45 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self._seed_urls = config.get_seed_urls()
+        self._config = config
+        self.urls = []
 
-    def _extract_url(self, article_bs: BeautifulSoup) -> str:
+    @staticmethod
+    def _extract_url(article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        all_links = []
+        all_links.extend(article_bs.find_all('a', class_="article-list__title"))
+        all_links.extend(article_bs.find_all('a', class_="article__embedded"))
+        all_links.extend(article_bs.find_all('a', class_="card__title"))
+        for link in all_links:
+            try:
+                address = link['href']
+            except KeyError:
+                continue
+            yield address
```

Comment:
solution is beautiful, but breaks typing that you cannot change. now you return iterator generic of type str, but typings require you to return a single string. rework.

` _extract_url(article_bs: BeautifulSoup)` is responsible for extracting single url from each article block